DigitalLearningSolutions.Web/ViewComponents/LogoViewComponent.cs

```diff
@@ -1,35 +1,26 @@
-namespace DigitalLearningSolutions.Web.ViewComponents
-{
-    using DigitalLearningSolutions.Data.Services;
-    using DigitalLearningSolutions.Web.Helpers;
-    using Microsoft.AspNetCore.Mvc;
-    using DigitalLearningSolutions.Web.ViewModels.Common;
-
-    public class LogoViewComponent : ViewComponent
-    {
-        private readonly ICentresService centresService;
-
-        public LogoViewComponent(ICentresService centresService)
-        {
-            this.centresService = centresService;
-        }
-
-        public IViewComponentResult Invoke()
-        {
-            var centreId = UserClaimsPrincipal.GetCustomClaimAsInt(CustomClaimTypes.UserCentreId);
-            if (centreId == null)
-            {
-                return View(new LogoViewModel(null));
-            }
-
-            var customLogo = centresService.GetCentreLogo(centreId.Value);
-            if (customLogo.LogoUrl == null)
-            {
-                return View(new LogoViewModel(null));
-            }
-
-            var model = new LogoViewModel(customLogo);
-            return View(model);
-        }
-    }
-}
+namespace DigitalLearningSolutions.Web.ViewComponents
+{
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.AspNetCore.Mvc;
+    using DigitalLearningSolutions.Web.ViewModels.Common;
+
+    public class LogoViewComponent : ViewComponent
+    {
+        private readonly ILogoService logoService;
+
+        public LogoViewComponent(ILogoService logoService)
+        {
+            this.logoService = logoService;
+        }
+
+        public IViewComponentResult Invoke(int? customisationId)
+        {
+            var centreId = UserClaimsPrincipal.GetCustomClaimAsInt(CustomClaimTypes.UserCentreId);

```

Comment:
There's a helper method for this, `User.GetCentreId` 🙂

Reply:
Ah I didn't realise that, thanks! 🙂 I had to cast types to make this work like this: `((ClaimsPrincipal) User).GetCentreId()`, I hope that's ok! Done in 1f0fb8c 🙂